feat(descriptor): add bip-86 template

reez opened the pull request:
<!-- Erase any parts of this template not applicable to your Pull Request. -->

### Description

This PR adds the descriptor template for bip-86.

<!-- Describe the purpose of this PR, what's being adding and/or fixed -->

### Links

[Merged `bdk` Pull Request adding BIP-86 template](https://github.com/bitcoindevkit/bdk/pull/1033)

[Docs for BIP-86 template](https://docs.rs/bdk/latest/bdk/descriptor/template/struct.Bip86.html)

### Notes to the reviewers

<!-- In this section you can include notes directed to the reviewers, like explaining why some parts
of the PR were done in a specific way -->

### Changelog notice

<!-- Notice the release manager should include in the release tag message changelog -->
<!-- See https://keepachangelog.com/en/1.0.0/ for examples -->

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

notmandatory commented:
Looks good! we should make a local swift package with this branch and try it out in your example ios wallet.

reez commented:
> Looks good! we should make a local swift package with this branch and try it out in your example ios wallet.

Can do!

thunderbiscuit reviewed (changes requested):
Tested. Looks good! Would you mind adding to the tests that start on line 266? Just for completeness since the other 3 templates are tested/showcased there.

I also think we can remove the comment on line 304; the templates are now showing correct coin types on testnet.

Otherwise ready to go IMO.

reez commented:
> Looks good! we should make a local swift package with this branch and try it out in your example ios wallet.

Just created a local swift package off of this branch and tried it out in the [iOS wallet](https://github.com/reez/BDKSwiftExampleWallet) creating and using a bip-86 wallet, and it worked great.

![Simulator Screenshot - iPhone 14 Pro - 2023-08-11 at 11 09 06](https://github.com/bitcoindevkit/bdk-ffi/assets/6657488/de4acf7c-2ef1-4cd9-a91f-f1ee12e7aac9)

reez commented:
> Tested. Looks good! Would you mind adding to the tests that start on line 266? Just for completeness since the other 3 templates are tested/showcased there.
> 
> I also think we can remove the comment on line 304; the templates are now showing correct coin types on testnet.
> 
> Otherwise ready to go IMO.

Definitely! The code is updated now, and tests pass ✅.
- Added tests for bip-86
- removed comment on line 304

  *Note*
  I changed line 307 
  `let template_private_84 = Descriptor::new_bip84(master, KeychainKind::External, Network::Testnet);` 
  to 
  `let template_private_84 = Descriptor::new_bip84(master.clone(), KeychainKind::External, Network::Testnet);` 
  just because of the ownership rules since I added the line 
  `let template_private_86 = Descriptor::new_bip86(master, KeychainKind::External, Network::Testnet);` 
  after that line. 
  The other quick option was to just add `.clone()` to both of those lines, but I kept things similar to how it was before and just made the last of those lines (`template_private_86`) not have `.clone()`.

thunderbiscuit reviewed (approved):
Tested ACK c3e8469686ff0c9aace45d0578e3d4162ea236a8.

thunderbiscuit commented:
@reez do you mind rebasing this? I'll merge right away. Sorry this took so long to review/get in.
